src/platform_impl/linux/window.rs

```diff
@@ -233,6 +242,43 @@ impl Window {
       signal_id.borrow_mut().replace(id);
     }
 
+    let mut win = Self::new_from_gtk_window(
```

Comment:
So for my immediate use case that would work, but I was thinking in the more general sense this would be a way to have interoperability with other ecosystems.

Like: maybe there's a library there that builds a window with various settings and returns it, and then the user wants to attach a webview to it.

In that case, if this method took the attributes and applied them, it could potentially undo or apply conflicting settings to the window.  The user could work around it but they'd have to be very careful to make all the attributes match what the library set where there's overlap, and it would make it more fragile if the upstream library changes or if Tao changes (to re-sync their attributes to the behaviors of both).

Why do you prefer the latter here? Are some of those attributes critical for correct webview operation? FWIW I'm happy to change this, but just want to make sure the discussion has covered all the issues first.

Reply:
I still think I'd prefer if we took the gtk window and applied the different attributes so it feels closer to what we would create, however I see you have a valid use-case about interoperability with other libraries, and so we could add both APIs.

Or actually we could go with only one API but apply no attributes at all, and just hook it to the event loop and so it will be the responsibility of the gtk window provider to apply the different attributes they need. This makes it a lot more predictable IMO.

Reply:
I like the one API idea, how were you thinking that would work with `WindowBuilder` in that case? Is one api = `new_from_gtk_window` only (named `new`) and the existing `new` would be replaced?

My understanding is that right now `WindowBuilder` calls `new` with attributes and results in similar behavior on all platforms for a given set of attributes. `Window::new_from_gtk_window` would only be called by linux developers who specifically need advanced gtk customization.  So if the attribute application doesn't go in `new` it'd need to be in some other platform-specific function that gets called by `WindowBuilder`, I presume.

Reply:
yeah I would only add `Window::new_from_gtk_window` which will only hook to the event loop, and return a tao `Window`, we won't provide a `WindowBuilder` variant nor we will change the behavior of `Window::new`

Reply:
Ah okay, got it.  So we'll have `new_from_gtk_window` which is like what was added here, and I'll revert the changes to `new` so it's exactly the same as before. Thanks for the clarification!

Reply:
Done.